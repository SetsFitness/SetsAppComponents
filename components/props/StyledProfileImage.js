import React from 'react';

type Props = {
  profileImage: any,
  type: string
};

/**
 * Creates the image based on the image and the type passed in as a prop.
 *
 * @param {*} image The image to display.
 * @param {string} type The
 * @return {*}
 */
const switchImageType = (image, type) => {
  switch (type) {
    case "Small":
      return <div className="u-avatar u-avatar--small" style={{backgroundImage: `url(${image})`}}/>;
    case "LargeCenter":
      return <div className="u-avatar u-avatar--large u-margin-bottom--neg2 u-margin-x--auto"
                  style={{backgroundImage: `url(${image})`}}/>
    default:
      return <div className="u-avatar u-avatar--small" style={{backgroundImage: `url(${image})`}}/>;
  }
};

/**
 * A styled profile image to switch through various ways of displaying and laying out a profile image.
 *
 * @param {Props} props The props passed into the component.
 * @return {*} The React JSX to display the component.
 * @constructor
 */
const StyledProfileImage = (props: Props) => switchImageType(props.profileImage, props.type);

export default StyledProfileImage;