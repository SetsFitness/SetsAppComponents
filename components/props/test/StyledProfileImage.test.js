import "../../../testing/SetTesting";
import React from 'react';
import { shallow } from 'enzyme';
import StyledProfileImage from "../StyledProfileImage";

it('renders without crashing', () => {
    const component = shallow(<StyledProfileImage profileImage={null} type={null}/>);
    expect(component).toMatchSnapshot();
});
