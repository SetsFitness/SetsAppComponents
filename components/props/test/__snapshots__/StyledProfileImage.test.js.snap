// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StyledProfileImage
    profileImage={null}
    type={null}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "u-avatar u-avatar--small",
      "style": Object {
        "backgroundImage": "url(null)",
      },
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "u-avatar u-avatar--small",
        "style": Object {
          "backgroundImage": "url(null)",
        },
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
